### PR TITLE
Refactor media area to ES6. Remove promise renderer

### PR DESCRIPTION
**Pull request by srallen**
For #3216, refactors the `media-area.cjsx` into ES6 and removes the `PromiseRenderer`. Since this file was technically a set of components, I broke them out, created a "controller" component to handle the data and a separate view component as its child. I also moved it into the lab folder since it is used by the `EditMediaPage` component and the survey task editor. `Thumbnail` remains in the components folder since it is used by talk.

This could use some tests, but I'd like to do that separately since this is already touching so many files. If you'd prefer tests coming in with this though, I can do that.

Staged at https://refactor-media-area.pfe-preview.zooniverse.org/

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [x] Did you deploy a staging branch?
- [ ] Did you get any type checking errors from [Babel Typecheck](https://github.com/codemix/babel-plugin-typecheck)

## Optional

- [x] If it's a new component, is it in ES6? Is it clear of warnings from ESLint?
- [x] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?

### CODE REVIEW

**Comment by srallen** [comment]
Just noticed I broke some drag and drop target styles. Gonna look at this morning.

**Comment by srallen** [comment]
Also, this [talk comment](https://www.zooniverse.org/talk/18/144354) seems to be about `504` and `412` errors being thrown on the `PUT` request on the workflow when there's a large batch of images dropped onto the target for upload. Gonna take a look at my refactor to see if that's still an issue with it.

**Comment by srallen** [comment]
I've fixed the drag and drop styling. For the talk comment, I moved the error div to be at the top of the media area to make sure it is seen. There were probably errors in the UI when the issue described in the Talk thread happened, but they couldn't be seen previously since there was enough media causing an overflow and scroll bar and the error div was originally at the bottom of all of that.

File: app/components/thumbnail.jsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        <div>
+          <video width="300" controls>
+            <source src={this.props.src} type="video/mp4" />
+           </video>
+        </div>
+      );
+    }
+
+    return (
+      <img alt="" {...this.props} src={src} {...dimensions} style={style} onError={this.handleError} />
```
Comment (mcbouslog) [inline app/components/thumbnail.jsx]: more for my understanding than a review note, what does destructuring `...this.props` give the `img` component we're not already setting individually? or more likely I'm misreading.

File: app/components/thumbnail.jsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    };
+  }
+
+  getThumbnailSrc({ origin, width, height, src }) {
+    const srcPath = src.split('//').pop();
+    return (`${origin}/${width}x${height}/${srcPath}`);
+  }
+
+  handleError() {
+    if (!this.state.failed) {
```
Comment (mcbouslog) [inline app/components/thumbnail.jsx]: when attempting to upload the FilterIcons (.svg's) from Ali's sample survey task via Project Builder, Media or Survey Task editor, console had 415 errors and the following error message:
```
thumbnail.jsx?bf6a:20 Uncaught TypeError: Cannot read property 'state' of null(…)
```

Uploading ReferencePhotos (.jpg's) worked great.

Comment (parrish) [inline app/components/thumbnail.jsx]: `this` is unbound here.  Adding `this.handleError = this.handleError.bind(this);` to the constructor fixes it

File: app/components/thumbnail.jsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      maxWidth: this.props.width,
+      maxHeight: this.props.height,
+    };
+
+    if (this.props.format === 'mp4') {
+      return (
+        <div>
+          <video width="300" controls>
+            <source src={this.props.src} type="video/mp4" />
+           </video>
```
Comment (parrish) [inline app/components/thumbnail.jsx]: Alignment

File: app/components/thumbnail.jsx (hunk 2 above)
Comment (parrish) [inline app/components/thumbnail.jsx]: That's from what I mentioned here: https://github.com/zooniverse/Panoptes-Front-End/pull/3330#discussion_r92707937

File: app/components/thumbnail.jsx (hunk 1 above)
Comment (parrish) [inline app/components/thumbnail.jsx]: It adds a bit of flexibility to the component (e.g. using arbitrary image attributes)

**Comment by srallen** [comment]
Thanks for catching those. @parrish, I've restaged for another review.

**Pull request merged**